wordpress-mobile/WordPress-Android
# Allow resuming the migration after logout from site address login

Fixes #17814

This PR makes it possible to resume the migration flow after logout when the user is logged in only with a site address.
It also adds the `Log out` button and functionality to the Migration Help screen, which mimics the existing logout functionality available by navigating back from the other migration screens.

Related to & possibly resulting in merge conflicts requiring manual resolution after merging PR:
- #17816

To test:
1. Uninstall all WordPress and Jetpack apps on your device
2. Build & install the WordPress app from this branch
3. Login into the WordPress app with a site address
4. Build & install the Jetpack app from this branch, using the same flavor as installed the WordPress app
5. **Expect** the app to start on the migration flow
6. Navigate back (this will make the app logout and show the Landing screen)
7. Force close & open the Jetpack app again
8. ✅ **Verify** that the migration flow resumes without errors
9. Tap `Need help?`
10. ✅ **Verify** that a `Log out` button is shown
11. Tap `Log out` (this will make the app logout and show the Landing screen)
12. Force close & open the Jetpack app again
13. ✅ **Verify** that the migration flow resumes without errors

## Regression Notes
1. Potential unintended areas of impact
   Logout from WPcom account when having WPorg site addresses as well.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual and unit testing.

4. What automated tests I added (or what prevented me from doing so)
   Added unit tests for the new logic in `HelpViewModelTest` and `JetpackMigrationViewModelTest`.

PR submission checklist:
